lib/engine/game/base.rb

```diff
@@ -1377,12 +1377,12 @@ def compute_stops(route, train = nil)
           stops, revenue = visits.combination(num_stops.to_i).map do |stops|
             # Make sure this set of stops is legal
             # 1) At least one stop must have a token (if enabled)
-            next if train.requires_token && stops.none? { |stop| stop.tokened_by?(route.corporation) }
+            next if train.requires_token && stops.none? { |stop| stop&.tokened_by?(route.corporation) }
 
             # 2) We can't ask for more revenue centers of a type than are allowed
             types_used = Array.new(distance.size, 0) # how many slots of each row are filled
 
-            next unless stops.all? do |stop|
+            next unless stops.flatten.all? do |stop|
```

Comment:
This bug is in revenue_stops().

`stops << sorted_stops.select { |stop| trans_siberian_hexes.include?(stop.hex) } if trans_siberian_bonus (sorted_stops)
`

should be

`stops.concat(sorted_stops.select { |stop| trans_siberian_hexes.include?(stop.hex) }) if trans_siberian_bonus?(sorted_stops)`

Reply:
amazing, thanks for finding the source of the bug!